integration/restValidation.ts

```diff
@@ -0,0 +1,215 @@
+            if (firstIndex !== -1) return -1;
+            firstIndex = index;
+        }
+    });
+    return firstIndex;
+};
+
+type CommitteesRaw = {
+    index: string;
+    slot: string;
+    validators: string[];
+};
+type Committees = {
+    index: number;
+    slot: number;
+    validators: number[];
+};
+const getCommitteesFactory = (baseUrl: string) => async (
+    validatorIndex: number,
+    blockSlot: number,
+    ignoreBefore?: number,
+): Promise<Committees[]> => {
+    const url = new URL(`/eth/v1/beacon/states/${blockSlot}/committees`, baseUrl);
+    const response = await axios.get(url.href);
+    const committees = response.data.data.map(({index, slot, validators}: CommitteesRaw) => ({
+        index: Number(index),
+        slot: Number(slot),
+        validators: validators.map(Number),
+    }));
+    return committees.filter(({validators, slot}: Committees) =>
+        validators.some((index) => index === validatorIndex && (slot > ignoreBefore || ignoreBefore === undefined)),
+    );
+};
+
+type ProposerRaw = {
+    pubkey: string;
+    // eslint-disable-next-line camelcase
+    validator_index: string;
+    slot: string;
+};
+type Proposer = {
+    index: number;
+    slot: number;
+    pubKey: string;
+};
+const getProposerFactory = (baseUrl: string) => async (
+    validatorIndex: number,
+    epoch: number,
+    ignoreBefore?: number,
+): Promise<Proposer[]> => {
+    const url = new URL(`/eth/v1/validator/duties/proposer/${epoch}`, baseUrl);
+    const response = await axios.get(url.href);
+    // eslint-disable-next-line camelcase,@typescript-eslint/camelcase
+    const proposers = response.data.data.map(({pubkey, validator_index, slot}: ProposerRaw) => ({
+        // eslint-disable-next-line camelcase,@typescript-eslint/camelcase
+        index: Number(validator_index),
+        slot: Number(slot),
+        pubKey: pubkey,
+    }));
+    return proposers.filter(
+        ({slot, index}: Proposer) => validatorIndex === index && (slot > ignoreBefore || ignoreBefore === undefined),
+    );
+};
+
+const isBlockProposedFactory = (baseUrl: string) => async (slot: number): Promise<boolean> => {
+    const url = new URL(`/eth/v1/beacon/blocks/${slot}`, baseUrl);
+    try {
+        const response = await axios.get(url.href);
+        if (Number(response.data.data.message.slot) === slot) {
+            return true;
+        }
+    } catch {
+        return false;
+    }
+    return false;
+};
+
+export const restValidation = ({
+    baseUrl,
+    getValidatorPrivateKey,
+    limit,
+}: {
+    baseUrl: string;
+    getValidatorPrivateKey: () => Promise<SecretKey>;
+    limit: number;
+}): Promise<{
+    proposer: {
+        proposed: number;
+        delegated: number;
+    };
+    attestation: {
+        attestations: number;
+        delegated: number;
+    };
+}> =>
+    new Promise((resolve) => {
+        (async (): Promise<void> => {
+            await initBLS("herumi");
```

Comment:
```suggestion
            await initBLS("blst-native");
```
is faster

Reply:
wont fix for some reason i got
> Require stack:
> - /home/bernard/WebstormProjects/ChainGuardian/node_modules/@chainsafe/bls/lib/blst/secretKey.js
> - /home/bernard/WebstormProjects/ChainGuardian/node_modules/@chainsafe/bls/lib/blst/index.js
> - /home/bernard/WebstormProjects/ChainGuardian/node_modules/@chainsafe/bls/lib/index.js

Reply:
thats stack, but what's the error? 😄

Reply:
```sh
yarn run v1.22.5
$ node -r ts-node/register integration/lighthouse.ts
(node:3219949) UnhandledPromiseRejectionWarning: Error: Cannot find module '@chainsafe/blst'
Require stack:
- /home/bernard/WebstormProjects/ChainGuardian/node_modules/@chainsafe/bls/lib/blst/secretKey.js
- /home/bernard/WebstormProjects/ChainGuardian/node_modules/@chainsafe/bls/lib/blst/index.js
- /home/bernard/WebstormProjects/ChainGuardian/node_modules/@chainsafe/bls/lib/index.js
- /home/bernard/WebstormProjects/ChainGuardian/integration/lighthouse.ts
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
    at Function.Module._load (internal/modules/cjs/loader.js:667:27)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/bernard/WebstormProjects/ChainGuardian/node_modules/@chainsafe/bls/lib/blst/secretKey.js:10:27)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Object.require.extensions.<computed> [as .js] (/home/bernard/WebstormProjects/ChainGuardian/node_modules/ts-node/src/index.ts:851:44)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
(node:3219949) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:3219949) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Done in 4.70s.
```

Reply:
ahh, yarn add @chainsafe/blst it's optional dependency. 

Not sure if that works with electron when compiled into binary.
Let's use herumi and maybe open issue to experiment adding blst.

Reply:
think it is problem in `ts-node/register` like i needed to add `eventsource` polyfill to make this tester work (and some other stuff)

Reply:
For error you posted `yarn add @chainsafe/blst` should work.

As for EventSource not sure if you even need it if you use CgEth2BeaconApi

Reply:
its used under `CgEth2EventsApi` :man_shrugging:

Reply:
how did it work before then? 😄

Reply:
probably magic :man_shrugging: JavaScript magic :mage:

Reply:
> For error you posted `yarn add @chainsafe/blst` should work.
> 
> As for EventSource not sure if you even need it if you use CgEth2BeaconApi

and works